### PR TITLE
[Rümker] New admin member page

### PR DESCRIPTION
**Pull request by philipperolet**
Related [Task](https://github.com/dust-tt/tasks/issues/79) and [Figma](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=1845%3A45444&mode=design&t=xdHeHkDVt5eM5aoP-1)

The page was almost completely rewritten to accomodate the figma cleanly. The code is stable, there are tweaks cf pending issues below but it won't change much => I think it's ripe for review

You can test it on https://front-edge.dust.tt/w/0ec9852c2f/members

Did what I could to make it ~ok on mobile

Pending topics on product / design:

- style of the hover, style of invitations modal:  @Duncid r?
- modals: awful notification to user: window.alert on fail, reload on success 
  - working on it, reloading is ~necessary but a nice notification after could be done => @Duncid would love your input here
- revoked members and revoked invitations will not be shown in this first iteration
  - or I can quickly add a checkbox "Show revoked" below the search bar (but then actionable or not?)
- Avatar image issue that i'm investigating

### CODE REVIEW

**Comment by spolu** [comment]
Looks great.

Only thing is that the modal has an animation problem for me. It comes from the right but over shoot to the left and wiggle back in place

File: front/pages/w/[wId]/members/index.tsx [hunk 1]
```diff
@@ -56,66 +70,315 @@ export default function WorkspaceAdmin({
...
+    <Modal
+      isOpen={showModal}
+      onClose={onClose}
+      hasChanged={false}
+      title="Invite new users"
+      type="right-side"
+    >
+      <div className="mt-6 flex flex-col gap-6 px-2 text-sm">
+        <Page.P>
+          Send an email to invite a new user to your workspace. They will be
```
Comment (PopDaph) [inline front/pages/w/[wId]/members/index.tsx:340]: Wording sounds weird to me but feel free to discard. Proposal to rename from 

> Send an email to invite a new user to your workspace. They will be able to join your workspace by clicking on the link in the email.

to: 

> Invite a new user to your workspace. They will receive an email with a link to  join your workspace.

File: front/pages/w/[wId]/members/index.tsx [hunk 2]
```diff
@@ -129,343 +392,257 @@ export default function WorkspaceAdmin({
...
+function InviteSettingsModal({
+  showModal,
+  onClose,
+  owner,
+}: {
+  showModal: boolean;
+  onClose: () => void;
+  owner: WorkspaceType;
+}) {
+  const [domainUpdating, setDomainUpdating] = useState(false);
```
Comment (PopDaph) [inline front/pages/w/[wId]/members/index.tsx:414]: NIT: set the type for all your `useState`?

`  const [domainUpdating, setDomainUpdating] = useState<boolean>(false);`

File: front/pages/w/[wId]/members/index.tsx [hunk 3]
```diff
@@ -129,343 +392,257 @@ export default function WorkspaceAdmin({
...
+      window.alert("Failed to update workspace.");
+      setDomainUpdating(false);
     } else {
-      await mutate(`/api/w/${owner.sId}/invitations`);
+      // We perform a full refresh so that the Workspace name updates and we get a fresh owner
+      // object so that the formValidation logic keeps working.
+      window.location.reload();
+    }
+  }
+  function validDomain() {
```
Comment (PopDaph) [inline front/pages/w/[wId]/members/index.tsx:475]: NIT: add types for all your functions too 🙏🏻 ?

File: front/pages/w/[wId]/members/index.tsx [hunk 4]
```diff
@@ -129,343 +392,257 @@ export default function WorkspaceAdmin({
...
+    <Modal
+      isOpen={showModal}
+      onClose={onClose}
+      hasChanged={false}
+      title="Revoke invitation"
+      type="right-side"
+    >
+      <div className="mt-6 flex flex-col gap-6 px-2">
+        <div>
+          Revoke invitation of user with email{" "}
```
Comment (PopDaph) [inline front/pages/w/[wId]/members/index.tsx:516]: NIT: not sure, but shouldn't it be "Revoke invitation **for** user with email"?

File: front/pages/w/[wId]/members/index.tsx [hunk 5]
```diff
@@ -129,343 +392,257 @@ export default function WorkspaceAdmin({
...
+
+function RevokeInvitationModal({
+  showModal,
+  onClose,
+  invitation,
+  owner,
+}: {
+  showModal: boolean;
+  onClose: () => void;
+  invitation: MembershipInvitationType | null;
```
Comment (PopDaph) [inline front/pages/w/[wId]/members/index.tsx:501]: How come invitation can be null here?

File: front/pages/w/[wId]/members/index.tsx [hunk 6]
```diff
@@ -129,343 +392,257 @@ export default function WorkspaceAdmin({
...
+            <Button
+              variant="primaryWarning"
+              label="Revoke member access"
+              size="sm"
+              onClick={() => handleMemberRoleChange(member, "none")}
+            />
+          </div>
+          <Page.P>
+            Deleting a member will remove them from the workspace. They will be
+            able to rejoin if they have an invitation link.
```
Comment (PopDaph) [inline front/pages/w/[wId]/members/index.tsx:626]: NIT: should we say "They will be able to rejoin if the invitation link is activated for a domain and they still have access to an email address for this domain." ? or too complex?

**Comment by philipperolet** [comment]
> Looks great.
> 
> Only thing is that the modal has an animation problem for me. It comes from the right but over shoot to the left and wiggle back in place

Addressed in sparkle, and bumped the sparkle version in front accordingly

File: front/package-lock.json [hunk 7]
```diff
@@ -9801,6 +9796,12 @@
         "node": ">= 0.8.0"
       }
     },
+    "node_modules/optionator/node_modules/fast-levenshtein": {
```
Comment (PopDaph) [inline front/package-lock.json:9799]: Do you know why this change?? 🤔

**Review by PopDaph — DISMISSED** [review]
Thanks for deploying on edge it's super convenient for reviewers! 

No blocker comments are mostly nits. Before merging I would just add the missing types and handle the modal animation issue + the comment on key. 

The new design looks really cool 🤩

File: front/pages/w/[wId]/members/index.tsx [hunk 8]
```diff
@@ -56,66 +70,315 @@ export default function WorkspaceAdmin({
...
+              size="sm"
+              icon={PlusIcon}
+              onClick={() => setInviteEmailModalOpen(true)}
+            />
+          </div>
+        </div>
+        <div>
+          {displayList.map((elt, i) => (
+            <div
+              key={i}
```
[the remark this answers is not in the document]
Comment (philipperolet) [reply front/pages/w/[wId]/members/index.tsx:246]: Good catch! :pray:

File: front/pages/w/[wId]/members/index.tsx (hunk 2 above)
Comment (philipperolet) [reply front/pages/w/[wId]/members/index.tsx:414]: Here I would argue that it's fine not to set the type for native types (boolean, string, number) since it's unambiguous from the set value so you know it when you see the type

File: front/pages/w/[wId]/members/index.tsx (hunk 3 above)
Comment (philipperolet) [reply front/pages/w/[wId]/members/index.tsx:475]: here OTOH I agree since you can't infer the return type without reading the function

File: front/pages/w/[wId]/members/index.tsx (hunk 4 above)
Comment (philipperolet) [reply front/pages/w/[wId]/members/index.tsx:516]: Well, ok :)
image

File: front/pages/w/[wId]/members/index.tsx (hunk 5 above)
Comment (philipperolet) [reply front/pages/w/[wId]/members/index.tsx:501]: Because when the page is loaded, before the user decides to hover on an invitation in the list, there is no invitation to pass to the modal (see corresponding useState)

**Review by Duncid — DISMISSED** [review]

File: front/pages/w/[wId]/members/index.tsx [hunk 9]
```diff
@@ -56,66 +70,326 @@ export default function WorkspaceAdmin({
...
-  const { invitations, isInvitationsLoading } = useWorkspaceInvitations(owner);
+  function MemberList({
+    members,
+    invitations,
+  }: {
+    members: UserType[];
+    invitations: MembershipInvitationType[];
+  }) {
+    const COLOR_FOR_ROLE: { [key: string]: "warning" | "amber" | "emerald" } = {
+      admin: "warning",
```
[the remark this answers is not in the document]
Comment (philipperolet) [reply front/pages/w/[wId]/members/index.tsx:162]: Current chip colors are "emerald" | "amber" | "slate" | "violet" | "warning" | "sky" | "pink" | "indigo" | "action" => we can add red, but I suggest picking from an existing one (the more colors, the more risk to use too many colors). 

Loosely held so if you really think red is needed here I'll add it to the chip in sparkle NP
